### ISSUE TITLE
xUser: unit tests appear not to mock functions

### ISSUE THREAD

**Issue by gaelicWizard** [dsccommunity/xPSDesiredStateConfiguration#720]
It seems like the unit tests in xUser actually create/modify/remove users on the testing server? I've only started learning unit testing this past week, so forgive me if I'm just being silly. Should I try writing new unit tests for xUser like there are for xGroup?

**Comment by PlagueHO** [comment]
I agree @gaelicWizard - we should actually be mocking these calls as it is best practice to not allow unit tests to actually make changes to the underlying platform. What we could do is move these to integration tests and create new ones that mock these. Integration tests are allowed to make changes to the underlying platform.
